### PR TITLE
[k] linalg category ops fix

### PR DESCRIPTION
**Pull request by RobertLuobo**
Extract the linear-algebra functional subset of the dev_ops commit: linalg ops (linalg_cholesky, linalg_det, linalg_slogdet, linalg_ldl_factor(_ex), linalg_ldl_solve, linalg_lstsq, linalg_lu_factor(_ex), linalg_matrix_norm, linalg_cross, linalg_householder_product, linalg_solve_triangular, cholesky_inverse, cholesky_solve, lu_unpack, ormqr, _linalg_eigvals) and matmul/norm ops (addmm, addmm_, baddbmm, bmm, mm, mv, dot, _embedding_bag, eye_m, kron, linear_backward, mvlgamma, norm, renorm, trace, vector_norm) with vendor-specific XPU implementations, plus registration entries in ops/__init__.py.

### Progress

- [ ] Change is properly reviewed (1 reviewer required, 2 recommended).
- [ ] Change is responded to an issue.
- [ ] Change is fully covered by a UT.

### CODE REVIEW

File: src/flag_gems/runtime/backend/_kunlunxin/ops/_linalg_eigvals.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+
+
+def _linalg_eigvals(inp):
+    return torch.linalg.eig(inp).eigenvalues
```
Comment (Caeruleann) [inline src/flag_gems/runtime/backend/_kunlunxin/ops/_linalg_eigvals.py]: _linalg_eigvals is currently implemented entirely through torch.linalg.eig, so it bypasses FlagGems and contains no Triton/FlagGems device computation. Please implement this with the Kunlunxin backend or existing FlagGems operators instead of calling the native PyTorch operator.

File: src/flag_gems/runtime/backend/_kunlunxin/ops/linalg_lstsq.py [hunk 2]
```diff
@@ -0,0 +1,595 @@
...
+    row wider than the validated reduction tile). Routed through CPU so it does
+    not re-enter this override."""
+    res = torch.linalg.lstsq(A.cpu(), b.cpu(), rcond=rcond, driver=driver)
+    dev = A.device
+    return (
+        res.solution.to(dev),
+        res.residuals.to(dev),
+        res.rank.to(dev),
+        res.singular_values.to(dev),
+    )
```
Comment (Caeruleann) [inline src/flag_gems/runtime/backend/_kunlunxin/ops/linalg_lstsq.py]: This fallback moves the inputs to CPU and calls native torch.linalg.lstsq, which violates the no-native-fallback and host-compute requirements and introduces synchronous device transfers. Please implement these cases with Triton/existing FlagGems operators or explicitly reject unsupported cases.

File: src/flag_gems/runtime/backend/_kunlunxin/ops/linalg_matrix_norm.py [hunk 3]
```diff
@@ -0,0 +1,573 @@
...
+        raise RuntimeError(
+            f"linalg_matrix_norm: A must be at least 2-D, got shape {A.shape}"
+        )
+    dim = list(dim)
+    if len(dim) != 2:
+        raise RuntimeError(f"linalg_matrix_norm: dim must be a 2-tuple, got {dim}")
+    dim = [d % A.ndim for d in dim]
+    if dim[0] == dim[1]:
+        raise RuntimeError(
+            f"linalg_matrix_norm: dims must be different, got ({dim[0]}, {dim[1]})"
```
Comment (Caeruleann) [inline src/flag_gems/runtime/backend/_kunlunxin/ops/linalg_matrix_norm.py:529]: Please validate each dimension before applying modulo normalization. Currently out-of-range dimensions such as (2, 3) for a 2-D input are silently converted to (0, 1) instead of raising like ATen. Please add positive and negative out-of-range dimension tests.

File: src/flag_gems/runtime/backend/_kunlunxin/ops/linear_backward.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+from .mm import mm
+from .sum import sum_dim
+
+logger = logging.getLogger(__name__)
+
+
+def linear_backward(input, grad_output, weight, output_mask):
+    logger.debug("GEMS_KUNLUNXIN LINEAR_BACKWARD")
+    batch_dims = input.shape[:-1]
+    batch_size = input.numel() // input.shape[-1]
```
Comment (Caeruleann) [inline src/flag_gems/runtime/backend/_kunlunxin/ops/linear_backward.py]: input.numel() // input.shape[-1] raises ZeroDivisionError for valid zero-width inputs such as input=(batch, 0) and weight=(out_features, 0). Please derive the batch size without dividing by in_features and add zero-sized forward/backward parity tests.

File: src/flag_gems/runtime/backend/_kunlunxin/ops/_linalg_eigvals.py (hunk 1 above)
Comment (RobertLuobo) [reply src/flag_gems/runtime/backend/_kunlunxin/ops/_linalg_eigvals.py]: DONE

File: src/flag_gems/runtime/backend/_kunlunxin/ops/linear_backward.py (hunk 4 above)
Comment (RobertLuobo) [reply src/flag_gems/runtime/backend/_kunlunxin/ops/linear_backward.py]: image

The general triton op also ignore it. We just follow the given triton kernel.

File: src/flag_gems/runtime/backend/_kunlunxin/ops/linalg_lstsq.py (hunk 2 above)
Comment (RobertLuobo) [reply src/flag_gems/runtime/backend/_kunlunxin/ops/linalg_lstsq.py]: DONE.

File: src/flag_gems/runtime/backend/_kunlunxin/ops/linalg_matrix_norm.py (hunk 3 above)
Comment (RobertLuobo) [reply src/flag_gems/runtime/backend/_kunlunxin/ops/linalg_matrix_norm.py:529]: Testcases would be provided by flagos-flagGems.

Comment (RobertLuobo) [reply src/flag_gems/runtime/backend/_kunlunxin/ops/linalg_matrix_norm.py:529]: @Caeruleann Please help me run test： /test linalg_matrix_norm

File: src/flag_gems/runtime/backend/_kunlunxin/ops/linear_backward.py (hunk 4 above)
Comment (RobertLuobo) [reply src/flag_gems/runtime/backend/_kunlunxin/ops/linear_backward.py]: DONE

**Review by Caeruleann — APPROVED** [review]
